From c7b22062b19a51b405667c0e8002b19ca16a84e9 Mon Sep 17 00:00:00 2001
From: Julien Gilli <julien.gilli@joyent.com>
Date: Tue, 13 Feb 2018 14:04:59 -0800
Subject: [PATCH] TRITON-130 mounting non-existing volume with same name from
 more than one VM fails Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
 Approved by: Josh Wilsdon <jwilsdon@joyent.com>

---
 lib/workflows/nfs-volumes/index.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/workflows/nfs-volumes/index.js b/lib/workflows/nfs-volumes/index.js
index f443c65..962784b 100644
--- a/lib/workflows/nfs-volumes/index.js
+++ b/lib/workflows/nfs-volumes/index.js
@@ -292,7 +292,10 @@ function provisionNfsVolumes(job, cb) {
                             name: volumeName,
                             owner_uuid: job.params.owner_uuid,
                             predicate: JSON.stringify({
-                                eq: ['state', 'ready']
+                                or: [
+                                    {eq: ['state', 'ready']},
+                                    {eq: ['state', 'creating']}
+                                ]
                             })
                         }, function onListVolumes(listVolumesErr, volumes) {
                             var loadedVolume;
-- 
2.21.0

